From 39c02c1d0abc592404db8ae25271b33352f31859 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 22 Sep 2016 13:32:06 -0500
Subject: [PATCH 13/35] Revert "drm/tilcdc: Change tilcdc_crtc_page_flip() to
 tilcdc_crtc_update_fb()"

This reverts commit bcf7ab1c48206234412ec63c692b41507bbbe4e2.
---
 drivers/gpu/drm/tilcdc/tilcdc_crtc.c  | 5 +++--
 drivers/gpu/drm/tilcdc/tilcdc_drv.h   | 5 +++--
 drivers/gpu/drm/tilcdc/tilcdc_plane.c | 5 +++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/tilcdc/tilcdc_crtc.c b/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
index 8accc90..3b3fed5 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
@@ -225,9 +225,10 @@ static void tilcdc_crtc_destroy(struct drm_crtc *crtc)
 	drm_flip_work_cleanup(&tilcdc_crtc->unref_work);
 }
 
-int tilcdc_crtc_update_fb(struct drm_crtc *crtc,
+int tilcdc_crtc_page_flip(struct drm_crtc *crtc,
 		struct drm_framebuffer *fb,
-		struct drm_pending_vblank_event *event)
+		struct drm_pending_vblank_event *event,
+		uint32_t page_flip_flags)
 {
 	struct tilcdc_crtc *tilcdc_crtc = to_tilcdc_crtc(crtc);
 	struct drm_device *dev = crtc->dev;
diff --git a/drivers/gpu/drm/tilcdc/tilcdc_drv.h b/drivers/gpu/drm/tilcdc/tilcdc_drv.h
index 04ca43a..bfd0e2a 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_drv.h
+++ b/drivers/gpu/drm/tilcdc/tilcdc_drv.h
@@ -172,9 +172,10 @@ void tilcdc_crtc_set_simulate_vesa_sync(struct drm_crtc *crtc,
 int tilcdc_crtc_mode_valid(struct drm_crtc *crtc, struct drm_display_mode *mode);
 int tilcdc_crtc_max_width(struct drm_crtc *crtc);
 void tilcdc_crtc_disable(struct drm_crtc *crtc);
-int tilcdc_crtc_update_fb(struct drm_crtc *crtc,
+int tilcdc_crtc_page_flip(struct drm_crtc *crtc,
 		struct drm_framebuffer *fb,
-		struct drm_pending_vblank_event *event);
+		struct drm_pending_vblank_event *event,
+		uint32_t page_flip_flags);
 
 int tilcdc_plane_init(struct drm_device *dev, struct drm_plane *plane);
 
diff --git a/drivers/gpu/drm/tilcdc/tilcdc_plane.c b/drivers/gpu/drm/tilcdc/tilcdc_plane.c
index 41911e3..d5635d6 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_plane.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_plane.c
@@ -101,9 +101,10 @@ static void tilcdc_plane_atomic_update(struct drm_plane *plane,
 	if (WARN_ON(!state->fb || !state->crtc->state))
 		return;
 
-	tilcdc_crtc_update_fb(state->crtc,
+	tilcdc_crtc_page_flip(state->crtc,
 			      state->fb,
-			      state->crtc->state->event);
+			      state->crtc->state->event,
+			      0);
 }
 
 static const struct drm_plane_helper_funcs plane_helper_funcs = {
-- 
2.9.3

